lib/index.js

```diff
@@ -0,0 +1,56 @@
+// Knex.js  0.8.0
+// --------------
+//     (c) 2014 Tim Griesser
+//     Knex may be freely distributed under the MIT license.
+//     For details and documentation:
+//     http://knexjs.org
+
+'use strict';
+
+var Raw            = require('./raw')
+var warn           = require('./helpers').warn
+var Client         = require('./client')
+
+var makeClient     = require('./util/make-client')
+var makeKnex       = require('./util/make-knex')
+var assign         = require('lodash/object/assign')
+
+function Knex(config) {
+  if (typeof config === 'string') {
+    return new Knex(assign(parseUrl(config), arguments[2]))
+  }
+  var Dialect;
+  if (arguments.length === 0 || (!config.client && !config.dialect)) {
+    Dialect = makeClient(Client)
+  } else {
+    var clientName = config.client || config.dialect
+    Dialect = makeClient(require('./dialects/' + (aliases[clientName] || clientName) + '/index.js'))
```

Comment:
it doesn't actually end up working in the browser, looking at the list of files looks like bluebird and lodash are the 2 main ones to slim the size

```

/usr/local/share/npm/lib/node_modules/browserify/node_modules/process/browser.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/events/events.js
/Users/cmetcalf/projects/knex/lib/functionhelper.js
/Users/cmetcalf/projects/knex/lib/query/methods.js
/Users/cmetcalf/projects/knex/lib/query/joinclause.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/punycode/punycode.js
/usr/local/share/npm/lib/node_modules/browserify/lib/_empty.js
/Users/cmetcalf/projects/knex/node_modules/inherits/inherits_browser.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/path-browserify/index.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/querystring-es3/encode.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/querystring-es3/decode.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/querystring-es3/index.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/url/url.js
/Users/cmetcalf/projects/knex/lib/util/parse-connection.js
/Users/cmetcalf/projects/knex/node_modules/lodash/index.js
/Users/cmetcalf/projects/knex/lib/transaction.js
/Users/cmetcalf/projects/knex/lib/schema/builder.js
/Users/cmetcalf/projects/knex/lib/schema/tablebuilder.js
/Users/cmetcalf/projects/knex/lib/schema/columnbuilder.js
/Users/cmetcalf/projects/knex/lib/schema/tablecompiler.js
/Users/cmetcalf/projects/knex/lib/schema/columncompiler.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseToString.js
/Users/cmetcalf/projects/knex/node_modules/lodash/utility/uniqueId.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/buffer/node_modules/is-array/index.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseCopy.js
/Users/cmetcalf/projects/knex/node_modules/lodash/function/restParam.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/assert/assert.js
/Users/cmetcalf/projects/knex/lib/query/builder.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/browser-resolve/empty.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/buffer/node_modules/ieee754/index.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/escape-string-regexp/index.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/ansi-styles/index.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/buffer/node_modules/base64-js/lib/b64.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/buffer/index.js
/Users/cmetcalf/projects/knex/lib/query/string.js
/Users/cmetcalf/projects/knex/lib/interface.js
/Users/cmetcalf/projects/knex/lib/raw.js
/Users/cmetcalf/projects/knex/node_modules/debug/node_modules/ms/index.js
/Users/cmetcalf/projects/knex/node_modules/debug/debug.js
/Users/cmetcalf/projects/knex/node_modules/debug/browser.js
/Users/cmetcalf/projects/knex/lib/migrate/index.js
/Users/cmetcalf/projects/knex/lib/schema/helpers.js
/Users/cmetcalf/projects/knex/lib/dialects/websql/transaction.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/supports-color/index.js
/Users/cmetcalf/projects/knex/lib/util/make-client.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/arrayEach.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/isObject.js
/Users/cmetcalf/projects/knex/node_modules/lodash/utility/identity.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/bindCallback.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/initCloneArray.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/arrayCopy.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/initCloneObject.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/isLength.js
/Users/cmetcalf/projects/knex/node_modules/lodash/utility/constant.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/isObjectLike.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/isIndex.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/util/support/isBufferBrowser.js
/Users/cmetcalf/projects/knex/node_modules/mkdirp/index.js
/Users/cmetcalf/projects/knex/lib/seed/index.js
/Users/cmetcalf/projects/knex/lib/util/make-knex.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseCreate.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/isArray.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/isTypedArray.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/toObject.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/getSymbols.js
/Users/cmetcalf/projects/knex/node_modules/lodash/string/escapeRegExp.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/isNative.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseAssign.js
/Users/cmetcalf/projects/knex/lib/query/compiler.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseIsFunction.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/isFunction.js
/Users/cmetcalf/projects/knex/lib/schema/compiler.js
/Users/cmetcalf/projects/knex/node_modules/lodash/support.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseProperty.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/getLength.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/isIterateeCall.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/createAssigner.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/strip-ansi/node_modules/ansi-regex/index.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/strip-ansi/index.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/has-ansi/node_modules/ansi-regex/index.js
/Users/cmetcalf/projects/knex/node_modules/chalk/node_modules/has-ansi/index.js
/Users/cmetcalf/projects/knex/node_modules/chalk/index.js
/Users/cmetcalf/projects/knex/lib/helpers.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/context.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/bind.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/synchronous_inspection.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/any.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/each.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/filter.js
/Users/cmetcalf/projects/knex/node_modules/pool2/node_modules/hashmap/hashmap.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/node_modules/inherits/inherits_browser.js
/Users/cmetcalf/projects/knex/node_modules/pool2/node_modules/double-ended-queue/js/deque.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/lib/_stream_transform.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/lib/_stream_passthrough.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/lib/_stream_duplex.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/lib/_stream_writable.js
/Users/cmetcalf/projects/knex/node_modules/pool2/lib/pool.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/util/util.js
/Users/cmetcalf/projects/knex/node_modules/pool2/lib/cluster.js
/Users/cmetcalf/projects/knex/node_modules/pool2/index.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/node_modules/isarray/index.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/bufferClone.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/initCloneByTag.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseClone.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/cloneDeep.js
/Users/cmetcalf/projects/knex/node_modules/lodash/lang/isArguments.js
/Users/cmetcalf/projects/knex/node_modules/lodash/object/keysIn.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/shimKeys.js
/Users/cmetcalf/projects/knex/node_modules/lodash/object/keys.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/assignWith.js
/Users/cmetcalf/projects/knex/node_modules/lodash/object/assign.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/util.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/thenables.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/promise_array.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/debuggability.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/captured_trace.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/catch_filter.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/progress.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/promise_resolver.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/method.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/direct_resolve.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/finally.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/join.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/map.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/cancel.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/using.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/generators.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/nodeify.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/props.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/call_get.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/race.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/reduce.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/settle.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/some.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/timers.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/inherits/inherits_browser.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseSlice.js
/Users/cmetcalf/projects/knex/node_modules/lodash/array/last.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/promisify.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/es5.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/errors.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/queue.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/node_modules/core-util-is/lib/util.js
/Users/cmetcalf/projects/knex/lib/dialects/sqlite3/schema/compiler.js
/Users/cmetcalf/projects/knex/lib/dialects/sqlite3/schema/columncompiler.js
/Users/cmetcalf/projects/knex/lib/dialects/sqlite3/schema/tablecompiler.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseIsMatch.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/isStrictComparable.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseMatches.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/basePropertyDeep.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseGet.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/isKey.js
/Users/cmetcalf/projects/knex/node_modules/lodash/utility/property.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/createBaseFor.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseFor.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseForOwn.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/toPath.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/writable.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/duplex.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/node_modules/string_decoder/index.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/lib/_stream_readable.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/schedule.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/async.js
/Users/cmetcalf/projects/knex/node_modules/bluebird/js/main/promise.js
/Users/cmetcalf/projects/knex/lib/promise.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/equalArrays.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/equalByTag.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/equalObjects.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseIsEqualDeep.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseIsEqual.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseMatchesProperty.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseCallback.js
/Users/cmetcalf/projects/knex/node_modules/lodash/object/transform.js
/Users/cmetcalf/projects/knex/lib/formatter.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/lib/_stream_passthrough.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/passthrough.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/arrayMap.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/node_modules/core-util-is/lib/util.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/lib/_stream_transform.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/transform.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/lib/_stream_writable.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/lib/_stream_duplex.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/isarray/index.js
/Users/cmetcalf/projects/knex/lib/dialects/sqlite3/query/compiler.js
/Users/cmetcalf/projects/knex/lib/dialects/sqlite3/schema/ddl.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/string_decoder/index.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/lib/_stream_readable.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/readable-stream/readable.js
/usr/local/share/npm/lib/node_modules/browserify/node_modules/stream-browserify/index.js
/Users/cmetcalf/projects/knex/node_modules/readable-stream/readable.js
/Users/cmetcalf/projects/knex/lib/runner.js
/Users/cmetcalf/projects/knex/lib/client.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/createBaseEach.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseEach.js
/Users/cmetcalf/projects/knex/node_modules/lodash/internal/baseMap.js
/Users/cmetcalf/projects/knex/node_modules/lodash/collection/map.js
/Users/cmetcalf/projects/knex/node_modules/lodash/collection/pluck.js
/Users/cmetcalf/projects/knex/lib/dialects/sqlite3/index.js
/Users/cmetcalf/projects/knex/lib/dialects/websql/index.js
/Users/cmetcalf/projects/knex/lib/index.js
/Users/cmetcalf/projects/knex
```